scripts/globals/summon.lua

```diff
@@ -10,112 +10,278 @@ function getSummoningSkillOverCap(avatar)
         skill:setMsg(tpz.msg.basic.SKILL_MISS)
-
-        -- some hits hit
     else
-        target:wakeUp()
-    end
+        -- https://www.bg-wiki.com/bg/Critical_Hit_Rate
+        -- Crit rate has a base of 5% and no cap, 0-100% are valid
+        -- Dex contribution to crit rate is capped and works in tiers
+        local baseCritRate = 5
+        local maxCritRate = 1 -- 100%
+        local minCritRate = 0 -- 0%
+
+        local critRate = baseCritRate + getDexCritRate(avatar, target) + avatar:getMod(tpz.mod.CRITHITRATE)
+        critRate = critRate / 100
+        critRate = utils.clamp(critRate, minCritRate, maxCritRate)
+        
+        local weaponDmg = avatar:getWeaponDmg()
+
+        local fSTR = getAvatarFSTR(weaponDmg, avatar:getStat(tpz.mod.STR), target:getStat(tpz.mod.VIT))
+
+        -- https://www.bg-wiki.com/bg/PDIF
+        -- https://www.bluegartr.com/threads/127523-pDIF-Changes-(Feb.-10th-2016)
+        local ratio = avatar:getStat(tpz.mod.ATT) / target:getStat(tpz.mod.DEF)
+        local cRatio = ratio
+
+        if shouldApplyLevelCorrection then
+            -- Mobs, Avatars and pets only get bonuses, no penalties (or they are calculated differently)
+            if levelDiff > 0 then
+                local correction = levelDiff * 0.05;
+                local cappedCorrection = math.min(correction, 1.9)
+                cRatio = cRatio + cappedCorrection
+            end
+        end
 
-    -- apply ftp bonus
-    if tpeffect == TP_DMG_BONUS then
-        finaldmg = finaldmg * avatarFTP(skill:getTP(), mtp100, mtp200, mtp300)
-    end
+        --Everything past this point is randomly computed per hit
 
-    returninfo.dmg = finaldmg
-    returninfo.hitslanded = hitslanded
+        numHitsProcessed = 0
 
-    return returninfo
-end
+        local targetHP = target:getHP()
 
--- minFstr = dSTR/4 + 0.5
--- maxFstr = dSTR/4 + 0.25
-function avatarFSTR(att_str, def_vit)
-    local dSTR = att_str - def_vit
-    return math.floor(dSTR / 4 + 0.5), math.floor(dSTR / 4 + 0.25)
-end
+        if firstHitLanded then
+            local wRatio = cRatio
+            local isCrit = math.random() < critRate
+            if isCrit then
+                wRatio = wRatio + 1
+            end
+            -- get a random ratio from min and max
+            local qRatio = getRandRatio(wRatio)
+
+            --Final pDif is qRatio randomized with a 1-1.05 multiplier
+            local pDif = qRatio * (1 + (math.random() * 0.05))
+            --Avatars have Crit Atk Bonus II for +8% crit dmg
+            if isCrit then
+                pDif = pDif * 1.08
+            end
+            finaldmg = avatarHitDmg(weaponDmg, fSTR, pDif) * dmgmod
+            numHitsProcessed = 1
+        end
+
+        -- If the first hit kills the mob other hits are not computed in retail
+        -- This can easily be seen by using BPs on low level mobs where the first hit kills them
+        while numHitsProcessed < numHitsLanded and finaldmg < targetHP do
```

Comment:
Problem with the `finaldmg < targetHP` check is that `AvatarPhysicalMove` is called _before_ defense adjustments for the defender `AvatarFinalAdjustments`, which can cause these cycles to break out early under the assumption the mob will be dead.

Reply:
Oh interesting okay, maybe down the road we need a final display damage and an internal total damage. Good catch 👍